File: .tmp_c4000/batch_04_go/3485_longest_common_prefix_of_k_strings_after_removal.go
// LeetCode 3485 - Longest Common Prefix of K Strings After Removal
// https://leetcode.com/problems/longest-common-prefix-of-k-strings-after-removal/

func longestCommonPrefix(words []string, k int) []int {
	n := len(words)
	ans := make([]int, n)
	for i := 0; i < n; i++ {
		// remove words[i], find max LCP among any k of remaining
		rest := []string{}
		for j, w := range words {
			if j != i {
				rest = append(rest, w)
			}
		}
		if len(rest) < k {
			ans[i] = 0
			continue
		}
		best := 0
		// brute combinations too heavy; sort and check windows of k
		sortStrings(rest)
		for j := 0; j+k-1 < len(rest); j++ {
			lcp := lcpOf(rest[j : j+k])
			if lcp > best {
				best = lcp
			}
		}
		ans[i] = best
	}
	return ans
}

func sortStrings(a []string) {
	for i := 0; i < len(a); i++ {
		for j := i + 1; j < len(a); j++ {
			if a[j] < a[i] {
				a[i], a[j] = a[j], a[i]
			}
		}
	}
}

func lcpOf(a []string) int {
	if len(a) == 0 {
		return 0
	}
	pref := a[0]
	for _, s := range a[1:] {
		i := 0
		for i < len(pref) && i < len(s) && pref[i] == s[i] {
			i++
		}
		pref = pref[:i]
		if len(pref) == 0 {
			return 0
		}
	}
	return len(pref)
}

File: .tmp_py4013/batch_13_dump/3485_longest_common_prefix_of_k_strings_after_removal.js
// LeetCode 3485 - Longest Common Prefix of K Strings After Removal
// https://leetcode.com/problems/longest-common-prefix-of-k-strings-after-removal/

var longestCommonPrefix = function(words, k) {
    const lcpOf = (a) => {
        if (!a.length) return 0;
        let pref = a[0];
        for (let t = 1; t < a.length; t++) {
            const s = a[t];
            let i = 0;
            while (i < pref.length && i < s.length && pref[i] === s[i]) i++;
            pref = pref.substring(0, i);
            if (!pref.length) return 0;
        }
        return pref.length;
    };
    const n = words.length;
    const ans = new Array(n);
    for (let i = 0; i < n; i++) {
        const rest = [];
        for (let j = 0; j < n; j++) if (j !== i) rest.push(words[j]);
        if (rest.length < k) { ans[i] = 0; continue; }
        rest.sort();
        let best = 0;
        for (let j = 0; j + k - 1 < rest.length; j++) {
            best = Math.max(best, lcpOf(rest.slice(j, j + k)));
        }
        ans[i] = best;
    }
    return ans;
};
